connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -110,14 +114,18 @@ public Map<String, ByteBuffer> performAssignment(String leaderId, String protoco
                 : CONNECT_PROTOCOL_V1;
 
         Long leaderOffset = ensureLeaderConfig(maxOffset, coordinator);
+        Map<String, ExtendedAssignment> assignments;
         if (leaderOffset == null) {
-            Map<String, ExtendedAssignment> assignments = fillAssignments(
+            assignments = fillAssignments(
                     memberConfigs.keySet(), Assignment.CONFIG_MISMATCH,
-                    leaderId, memberConfigs.get(leaderId).url(), maxOffset, Collections.emptyMap(),
-                    Collections.emptyMap(), Collections.emptyMap(), 0, protocolVersion);
-            return serializeAssignments(assignments);
+                    leaderId, memberConfigs.get(leaderId).url(), maxOffset,
+                    ClusterAssignment.EMPTY, 0, protocolVersion);
+        } else {
+            assignments = performTaskAssignment(leaderId, leaderOffset, memberConfigs, coordinator, protocolVersion);
         }
-        return performTaskAssignment(leaderId, leaderOffset, memberConfigs, coordinator, protocolVersion);
+        Map<String, ByteBuffer> result = serializeAssignments(assignments);
+        log.debug("Finished assignment");
```

Comment:
This works with `assignment's'` so maybe this?

```suggestion
        log.debug("Finished assignments");
```